docs/tracing.md

```diff
@@ -0,0 +1,306 @@
+spans for viewing and processing.
+
+To avoid having to insert a trace collector inside the VM image, the Kata
+Containers agent sends the trace spans generated by the agent to the [trace
+forwarder][trace-forwarder], which runs in the host environment. The forwarder
+then sends the spans to the trace collector, which also runs in the host
+environment. The communications between the agent and the trace forwarder are
+handled using a [`VSOCK`][vsock] channel.
+
+> **Note:**
+>
+> This design supports agent tracing without having to make changes to the
+> image, but also means that [custom images][osbuilder] can also benefit from
+> agent tracing.
+
+The following diagram summarises the architecture used to trace the
+Kata Containers agent:
+
+```
++-------------------------------------------+
+| Host                                      |
+|                                           |
+| +-----------+                             |
+| | Trace     |                             |
+| | Collector |                             |
+| +-----+-----+                             |
+|       ^                  +--------------+ |
+|       | spans            | Kata VM      | |
+| +-----+-----+            |              | |
+| | Kata      |    spans   |     +-----+  | |
+| | Trace     |<-----------------|Kata |  | |
+| | Forwarder |    VSOCK   |     |Agent|  | |
+| +-----------+    Channel |     +-----+  | |
+|                          +--------------+ |
++-------------------------------------------+
+```
+
+# Agent tracing terminology
+
+## Trace modes
+
+The agent supports two different tracing modes:
+
+| Trace mode | Description | Use-case | Setting | VM shutdown controller | Notes |
+|-|-|-|-|-|-|
+| Static | Traces from agent start to agent shutdown. | Holistic view of agent activities. Possibly more suited to short-running containers. | `trace_mode = "static"` | agent | Default trace mode. |
+| Dynamic | Traces begin when the ttRPC `StartTracing()` API is called and end when the corresponding `StopTracing()` API is called. | On-demand (partial) tracing. More appropriate for long-running containers. | `trace_mode = "dynamic"` | runtime | Only "isolated" type supported. |
+
+## Trace types
+
+Each trace mode is sub-divided into two different types:
+
+| Trace type | Description | Use-case | Setting | Notes |
+|-|-|-|-|-|
+| isolated | The traces only apply to the agent: the first span is generated at agent startup and the last at agent shutdown. | Observing agent lifespan. | `trace_type = "isolated"` | Default trace type. |
+| collated | In this mode, spans are associated with their Kata runtime initiated counterparts. | Understanding how the runtime and agent interact. | `trace_type = "collated"` | Also requires runtime tracing to be enabled in `configuration.toml` (`enable_tracing=true`). |
+
+# Prerequisites
+
+- You must have a trace collector running on the host.
+
+  The quickest and simplest method for doing this is to use the
```

Comment:
In that case, is the trace collector running within the docker image? (if so, that seems inconsistent to me with the mention that the collector runs on the host).